src/compas_model/model/group_node.py

```diff
@@ -48,3 +50,193 @@ def __init__(self, name=None, geometry=None, attributes=None, parent=None):
         # for debugging, the default name is the guid of th GroupNode
         # --------------------------------------------------------------------------
         self.name = name if name else str(self.guid)
+
+    # ==========================================================================
+    # Serialization
+    # ==========================================================================
+
+    @property
+    def data(self):
+        return {
+            "name": self.name,
```

Comment:
because you explicitly store the guid of the node when no name is provided, there is a lot of duplication.
better would be to do something like this:
```python
def __init__(...):
    ...
    self._name = None
    self.name = name

@property
def name(self):
    if not self._name:
        return str(self.guid)
    return self._name

@name.setter
def name(self, value):
    self._name = name
```